util/MultiplayerCommon.h

```diff
@@ -20,6 +21,132 @@ FO_COMMON_API extern const std::string SP_SAVE_FILE_EXTENSION;
+        std::string category = "";
+    };
+
+    /** \name Accessors */ //@{
+    bool Empty() const
+    { return m_game_rules.empty(); }
+    std::unordered_map<std::string, Rule>::const_iterator begin() const
+    { return m_game_rules.begin(); }
+    std::unordered_map<std::string, Rule>::const_iterator end() const
+    { return m_game_rules.end(); }
+
+    bool RuleExists(const std::string& name) const
+    { return m_game_rules.find(name) != m_game_rules.end(); }
+    bool RuleExists(const std::string& name, RuleType rule_type) const;
+    RuleType GetRuleType(const std::string& name) const;
+    bool RuleIsInternal(const std::string& name) const;
+
+    /** returns the description string for rule \a rule_name, or throws
+      * std::runtime_error if no such rule exists. */
+    const std::string& GetDescription(const std::string& rule_name) const;
+
+    /** returns the validator for rule \a rule_name, or throws
+      * std::runtime_error if no such rule exists. */
+    std::shared_ptr<const ValidatorBase> GetValidator(const std::string& rule_name) const;
+
+    /** returns all contained rules as name and value string pairs. */
+    std::vector<std::pair<std::string, std::string>> GetRulesAsStrings() const;
+
+    template <typename T>
+    T       Get(const std::string& name) const
+    {
+        auto it = m_game_rules.find(name);
+        if (it == m_game_rules.end())
+            throw std::runtime_error("GameRules::Get<>() : Attempted to get nonexistent rule \"" + name + "\".");
+        return boost::any_cast<T>(it->second.value);
+    }
+    //@}
+
+    /** \name Mutators */ //@{
+    /** adds a rule, optionally with a custom validator */
+    template <class T>
+    void    Add(const std::string& name, const std::string& description,
+                const std::string& category, T default_value,
+                bool engine_interal, const ValidatorBase& validator = Validator<T>())
+    {
+        auto it = m_game_rules.find(name);
+        if (it != m_game_rules.end())
+            throw std::runtime_error("GameRules::Add<>() : Rule " + name + " was added twice.");
+        m_game_rules[name] = Rule(RuleTypeForType(T()), name, default_value, default_value,
+                                  description, validator.Clone(), engine_interal, category);
+        DebugLogger() << "Added game rule named " << name << " with default value " << default_value;
+    }
+
+    template <typename T>
+    void    Set(const std::string& name, const T& value)
+    {
+        auto it = m_game_rules.find(name);
+        if (it == m_game_rules.end())
+            throw std::runtime_error("GameRules::Set<>() : Attempted to set nonexistent rule \"" + name + "\".");
+        it->second.SetFromValue(value);
+    }
+
+    void    SetFromStrings(const std::vector<std::pair<std::string, std::string>>& names_values);
+
+    /** Removes game rules that were added without being specified as
+        engine internal. */
+    void    ClearExternalRules();
```

Comment:
No, but it will be helpful when it becomes possible to change what rules are supported by changing the selected content directory.